src/khepri_fun.erl

```diff
@@ -916,22 +925,41 @@ pass2_process_instruction(
             Instruction
     end;
 pass2_process_instruction(
-  {wait_timeout, {f, OldLabel}, _} = Instruction,
-  #state{mfa_in_progress = {Module, _, _},
-         label_map = LabelMap}) ->
-    NewLabel = maps:get({Module, OldLabel}, LabelMap),
-    setelement(2, Instruction, {f, NewLabel});
-pass2_process_instruction(
-  {jump, {f, OldLabel}} = Instruction,
-  #state{mfa_in_progress = {Module, _, _},
-         label_map = LabelMap}) ->
-    NewLabel = maps:get({Module, OldLabel}, LabelMap),
-    setelement(2, Instruction, {f, NewLabel});
+  {wait_timeout, _, _} = Instruction, State) ->
+    replace_label(Instruction, 2, State);
 pass2_process_instruction(
   Instruction,
   _State) ->
     Instruction.
 
+replace_field_flags(Instruction, Pos) ->
+    {field_flags, FF} = element(Pos, Instruction),
+    setelement(Pos, Instruction, {field_flags, decode_field_flags(FF)}).
+
+decode_field_flags(0) ->
+    [];
+decode_field_flags(FieldFlag) ->
+    lists:filtermap(
+      fun
+          (little) ->
+              (FieldFlag band 16#02) == 16#02;
+          (signed) ->
+              (FieldFlag band 16#04) == 16#04;
+          (native) ->
+              (FieldFlag band 16#10) == 16#10
+      end, [signed, little, native]).
+
+replace_label(Instruction, Pos, #state{mfa_in_progress = {Module, _, _},
+                                       label_map = LabelMap}) ->
+    case element(Pos, Instruction) of
+        {f, 0} ->
+            %% Exception label, for branching
```

Comment:
I think we should expand the explanation in the comment and add link to https://github.com/erlang/otp/blob/master/lib/compiler/src/beam_validator.erl#L26-L32 (but a link to a specific commit or tag, so that it remains valid long after the code changes in the `master` branch).